src/opencl/mod.rs

```diff
@@ -185,13 +186,8 @@ impl Program {
         debug!("Creating OpenCL program from binary.");
         let context = Context::from_device(&device.device)?;
         let bins = vec![&bin[..]];
-        let mut program =
-            opencl3::program::Program::create_from_binary(&context, context.devices(), &bins)?;
-        if let Err(build_error) = program.build(context.devices(), "") {
-            let log = program.get_build_log(context.devices()[0])?;
-            return Err(GPUError::Opencl3(build_error, Some(log)));
-        }
-        let queue = CommandQueue::create(&context, context.default_device(), 0)?;
+        let program = opencl3::program::Program::create_and_build_from_binary(&context, &bins, "")?;
```

Comment:
There should still be an error message with the build log error in case it fails. That's important if you work edit an OpenCL kernel. Without that you wouldn't have any indication why the compilation failed.

Reply:
Corrected